api/communication-api/src/main/java/com/cognifide/aet/communication/api/metadata/Test.java

```diff
@@ -115,4 +116,25 @@ public String toString() {
         .add("name", name)
         .toString();
   }
+
+  public Optional<Url> getUrl(String urlName) {
+    Url urlToReturn = null;
+    for (Url url: this.urls) {
+      if(url.getName().equals(urlName)){
+        urlToReturn = url;
+        break;
+      }
+    }
+    return Optional.ofNullable(urlToReturn);
```

Comment:
This is just cosmetics and your implementation is perfectly fine. But you could also do this method body with streams:
```suggestion
    return urls.stream()
        .filter(url -> url.getName().equals(urlName))
        .findFirst();
```